doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+
+#### Primitive data types
+* `real` corresponds to the `double` data type in C++. Example literals are: `42.0`, `-0.42`, `.44`
+* `integer` corresponds to the `long` data type in C++. Example literals are: `42`, `-7`
+* `boolean` corresponds to the `bool` data type in C++. Its only literals are `true` and `false`
+* `string` corresponds to the `std::string` data type in C++. Example literals are: `"Bob"`, `""`, `"Hello World!"`
+* `void` corresponds to the `void` data type in C++. No literals are possible and this can only be used in the declaration of a function without return value.
+
+#### Physical units
+
+A physical unit in NESTML can be either a simple physical unit or a complex physical unit. A simple physical unit is composed of an optional magnitude prefix and the name of the unit.
+
+The following table lists seven base units, which can be used to specify any physical unit. This idea is based on [the SI units](https://en.wikipedia.org/wiki/International_System_of_Units).
+
+|Quantity | Unit Name | NESTML/SI unit |
+|---|---|---|
+| length              | metre    | m |
+| mass                | kilogram | kg |
+| time                | second   | s |
+| electric current    | ampere   | A |
+| temperature         | kelvin   | K |
+| amount of substance | mole     | mol |
+| luminous intensity  | candela  | cd |
+
+Any other physical unit can be expressed as a combination of these seven units. These other units are called derived units. NESTML provides the following derived units as built-in data types:
+
+|Quantity | Unit name | Definition | NESTML/SI unit |
+|---|---|---|---|
+| capacitance            | farad   | A<sup>2</sup> s<sup>4</sup> kg<sup>−1</sup> m<sup>−2</sup> | F |
+| electrical conductance | siemens | A<sup>2</sup> s<sup>3</sup> kg<sup>−1</sup> m<sup>−2</sup> | S |
+| electric charge        | coulomb | A s                                                        | C |
+| electric potential     | volt    | kg m<sup>2</sup> A<sup>−1</sup> s<sup>−3</sup>             | V |
+| electric resistance    | ohm     | kg m<sup>2</sup> A<sup>−2</sup> s<sup>−3</sup>             | ohm/&‌#x03A9; |
+| frequency              | hertz   | s<sup>−1</sup>                                             | Hz |
+
+| | | | |
+|---|---|---|---|
+|||| Bq |
+|||| Gy |
+|||| Sv |
+|||| J |
+|||| W  |
+|||| Wb |
+|||| N |
+|||| Pa |
+|||| kat |
+
+Units can have at most one of the following magnitude prefixes:
+
+|Factor | SI Name | NESTML prefix | Factor | SI Name | NESTML prefix
+|---|---|---|---|---|---|
+|10<sup>-1</sup>  | deci  | d  | 10<sup>1</sup>  | deca  | da |
+|10<sup>-2</sup>  | centi | c  | 10<sup>2</sup>  | hecto | h |
+|10<sup>-3</sup>  | milli | m  | 10<sup>3</sup>  | kilo  | k |
+|10<sup>-6</sup>  | micro | mu | 10<sup>6</sup>  | mega  | M |
+|10<sup>-9</sup>  | nano  | n  | 10<sup>9</sup>  | giga  | G |
+|10<sup>-12</sup> | pico  | p  | 10<sup>12</sup> | tera  | T |
+|10<sup>-15</sup> | femto | f  | 10<sup>15</sup> | peta  | P |
+
```

Comment:
@abigailm, @DimitriPlotnikov: I think we can safely leave out the ones below and just keep the ones above.